deepmd/utils/pair_tab.py

```diff
@@ -73,11 +163,18 @@ def _make_data(self):
                 dd *= self.hh
                 dtmp = np.zeros(stride)
                 for ii in range(self.nspline):
-                    dtmp[ii * 4 + 0] = 2 * vv[ii] - 2 * vv[ii + 1] + dd[ii] + dd[ii + 1]
+                    # check if vv is zero, if so, that's case 1, set all coefficients to 0,
+                    dtmp[ii * 4 + 0] = (
+                        2 * vv[ii] - 2 * vv[ii + 1] + dd[ii] + dd[ii + 1]
+                        if vv[ii] != 0
+                        else 0
+                    )
                     dtmp[ii * 4 + 1] = (
-                        -3 * vv[ii] + 3 * vv[ii + 1] - 2 * dd[ii] - dd[ii + 1]
+                        (-3 * vv[ii] + 3 * vv[ii + 1] - 2 * dd[ii] - dd[ii + 1])
+                        if vv[ii] != 0
+                        else 0
                     )
-                    dtmp[ii * 4 + 2] = dd[ii]
+                    dtmp[ii * 4 + 2] = dd[ii] if vv[ii] != 0 else 0
                     dtmp[ii * 4 + 3] = vv[ii]
                 data[
                     (t0 * self.ntypes + t1) * stride : (t0 * self.ntypes + t1) * stride
```

Comment:
if the table value at the beginning of the given spline is already `0` then, we set all 4 coefficients to `0`, to prevent negative energy.

Reply:
We decided to ignore this case, and allow negative energy.